demos/instr.h

```diff
@@ -77,4 +77,31 @@ inline void JumpToAfterSpeculation() {
   asm volatile("b afterspeculation");
 }
 #endif
+
+#if defined(__i386__) || defined(__x86_64__)
+__attribute__((always_inline))
+inline void EnforceAlignmentAndSerialize() {
+  asm volatile(
+      "pushf\n"
+#ifdef __i386__
+      "orl $0x00040000, (%%esp)\n"
+#else
+      "orl $0x00040000, (%%rsp)\n"
+#endif
+      "popf\n"
+      "cpuid\n":::"eax", "ebx", "ecx", "edx");
+}
```

Comment:
I think we'd be well-served to use less inline asm here. Let's break this up into three primitives that we can compose in regular C++:
- `cpuid` (which we've already built for other cases)
- `GetFlagsRegister()`
- `SetFlagsRegister()`

[The encoding of the `pushf` family](https://www.felixcloutier.com/x86/pushf:pushfd:pushfq) is a bit tricky -- the only thing 32-bit and 64-bit modes can agree on is getting the 16-bit `FLAGS` register -- so I'd recommend those latter two use `uintptr_t` to return and set the flags register value.

Reply:
Done. Separated the assemblies. Using uintptr_t brings a performance hit, but I used separate assembly constants instead.